server/src/main/java/org/elasticsearch/index/snapshots/IndexShardSnapshotStatus.java

```diff
@@ -272,6 +275,15 @@ public synchronized void addProcessedFiles(int count, long totalSize) {
         processedSize += totalSize;
     }
 
+    /**
+     * Updates the string explanation for what the snapshot is actively doing right now.
+     */
+    public void updateStatusDescription(String statusString) {
+        assert statusString != null;
+        assert statusString.isEmpty() == false;
```

Comment:
Nit: should really check the value passed into the constructor too in order to be certain that the `statusDescription` field is never null or empty.